crm_location/models/crm_lead.py

```diff
@@ -22,20 +22,22 @@ class CrmLead(models.Model):
     @api.depends("location_id")
     def _compute_partner_address_values(self):
         res = super()._compute_partner_address_values()
-        if self.location_id:
-            self.update(
-                {
-                    "zip": self.location_id.name,
-                    "city": self.location_id.city_id.name,
-                    "state_id": self.location_id.city_id.state_id,
-                    "country_id": self.location_id.city_id.country_id,
-                }
-            )
+        for lead in self:
+            if lead.location_id:
```

Comment:
And you avoid a big diff:

```suggestion
        for lead in self.filtered("location_id"):
```